backbone/src/test/java/org/researchstack/backbone/answerformat/AnswerFormatTest.java

```diff
@@ -39,4 +39,33 @@ public void testTestInvalidEmailAnswerFormat() throws Exception
         assertFalse(format.isAnswerValid("emailtest@.org"));
         assertFalse(format.isAnswerValid("12345"));
     }
+
+    @Test
+    public void testTextMinMaxRegexAnswerFormat()
+    {
+        TextAnswerFormat format = new TextAnswerFormat();
+        format.setValidationRegex("^\\w{4,16}$");
+        assertFalse(format.isAnswerValid("Abc"));               // less than 3 characters invalid
+        assertFalse(format.isAnswerValid("Abcd1234Abcd1234A")); // more than 17 characters
+        assertTrue(format.isAnswerValid("Abcd1234Abcd1234"));   // 16 characters
+        assertTrue(format.isAnswerValid("Abcd"));               // 4 characters is valid up to
+    }
+
+    @Test
+    public void testPasswordAnswerFormat()
+    {
+        PasswordAnswerFormat format = new PasswordAnswerFormat();
+        assertTrue(format.isAnswerValid("Abcd1234"));           // normal password valid
+        assertTrue(format.isAnswerValid("Abcd"));               // 4 characters is valid up to
+        assertTrue(format.isAnswerValid("Abcd1234Abcd1234"));   // 16 characters
```

Comment:
other requirements that bridge can require: includes 1 number, includes 1 lower case, 1 upper case, 1 symbol. These are optional tho.

Where did the max password length come from? I don't think that's a bridge thing.